src/realm/exec/clickbench.cpp

```diff
@@ -0,0 +1,379 @@
+                return Timestamp(epoch_days_fast(year, mon, day) * 86400 + hms, 0);
+            }
+        }
+    }
+    return Timestamp();
+}
+
+struct BufferedValues {
+    std::vector<std::string> buffer{256};
+    std::vector<FieldValues> values{256};
+};
+
+Mailbox<BufferedValues> mbx;
+Mailbox<BufferedValues> resp;
+
+static void parse_file(const char* filename)
+{
+    std::ifstream inp(filename);
+
+    auto buf = resp.receive();
+    auto str = buf->buffer.begin();
+    auto it = buf->values.begin();
+    auto end = buf->values.end();
+    while (std::getline(inp, *str)) {
+        char* tok = str->data();
+        for (FieldValue& val : *it) {
+            char* end = strchr(tok, '\t');
+            if (end) {
+                *end = '\0';
+            }
+            switch (val.col_key.get_type()) {
+                case col_type_Int:
+                    val.value = Mixed(int64_t(strtoll(tok, nullptr, 10)));
+                    break;
+                case col_type_String:
+                    val.value = strlen(tok) ? Mixed(tok) : Mixed();
+                    break;
+                    /*
+                                    case col_type_EnumString:
+                                        val.value = strlen(tok) ? Mixed(tok) : Mixed();
+                                        break;
+                    */
+                case col_type_Timestamp:
+                    val.value = Mixed(get_timestamp(tok));
+                    break;
+                default:
+                    break;
+            }
+            tok = end + 1;
+        }
+        ++it;
+        ++str;
+        if (it == end) {
+            mbx.send(buf);
+            buf = resp.receive();
+            str = buf->buffer.begin();
+            it = buf->values.begin();
+            end = buf->values.end();
+        }
+    }
+    buf->values.erase(it, end);
+    if (buf->values.size()) {
+        mbx.send(buf);
+    }
+    mbx.send(nullptr);
+}
+
+static void import(const char* filename)
+{
+    DBOptions options;
+    auto db = DB::create(make_in_realm_history(), "hits.realm");
+    create_table(db->start_write());
+    auto tr = db->start_write();
+    auto t = tr->get_table("Hits");
+    auto col_keys = t->get_column_keys();
+
+    std::cout << std::endl << "Reading data into realm" << std::endl;
+    auto time_start = std::chrono::high_resolution_clock::now();
+    BufferedValues buf1;
+    BufferedValues buf2;
+    for (auto& val : buf1.values) {
+        for (auto col : col_keys) {
+            val.insert(col, Mixed());
+        }
+    }
+    for (auto& val : buf2.values) {
+        for (auto col : col_keys) {
+            val.insert(col, Mixed());
+        }
+    }
+    resp.send(&buf1);
+    resp.send(&buf2);
+    std::thread parse_file_thread(parse_file, filename);
+
+    int buf_cnt = 0;
+    const int bufs_per_commit = 100;
+    while (auto buf = mbx.receive()) {
+        // t->create_objects(buf);
+        //        for (auto& val : buf->values) {
```

Comment:
Too many comments to actually import anything.

Reply:
ops my bad, it should now be OK, although the strings are not going to be compressed right now.